Marlin/src/gcode/feature/pause/M600.cpp

```diff
@@ -71,42 +101,101 @@ void GcodeSuite::M600() {
+      #endif 
+  #else
+    const float unload_length = parser.seen('U') ? parser.value_axis_units(E_AXIS) : 0
+      #ifdef FILAMENT_CHANGE_UNLOAD_LENGTH
+        - FILAMENT_CHANGE_UNLOAD_LENGTH;    
+      #endif 
+  #endif//end unload filament
+ 
   // Load filament
-  const float load_length = parser.seen('L') ? parser.value_axis_units(E_AXIS) : 0
-    #ifdef FILAMENT_CHANGE_LOAD_LENGTH
-      + FILAMENT_CHANGE_LOAD_LENGTH
-    #endif
-  ;
+  #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP)
+    float load_length = parser.seen('L') ? parser.value_axis_units(E_AXIS) : 0
+      #ifdef FILAMENT_CHANGE_LOAD_LENGTH
+	  + FILAMENT_CHANGE_LOAD_LENGTH ;
+      if (swap_spool_enabled) load_length = ADVANCED_PAUSE_SPOOL_SWAP_LOAD;
+      #endif 
+  #else    
+    const float load_length = parser.seen('L') ? parser.value_axis_units(E_AXIS) : 0
+      #ifdef FILAMENT_CHANGE_LOAD_LENGTH
+	    + FILAMENT_CHANGE_LOAD_LENGTH ;
+      #endif    
+  #endif //end load filament
 
   const int beep_count = parser.intval('B',
-    #ifdef FILAMENT_CHANGE_NUMBER_OF_ALERT_BEEPS
-      FILAMENT_CHANGE_NUMBER_OF_ALERT_BEEPS
+    #ifdef PAUSE_BEEPS
+      PAUSE_BEEPS
     #else
       -1
     #endif
   );
 
   const bool job_running = print_job_timer.isRunning();
 
-  if (pause_print(retract, z_lift, x_pos, y_pos, unload_length, beep_count, true)) {
-    wait_for_filament_reload(beep_count);
-    resume_print(load_length, ADVANCED_PAUSE_EXTRUDE_LENGTH, beep_count);
-  }
+  #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP)
+    
+    // Next tool 	
+    if (swap_spool_enabled ) {
+		
+      pause_print(retract, z_lift, x_pos, y_pos, unload_length, beep_count, false);
+	  
+	  #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP_COPY_FLOW)
+	  // Same flow after tool change 'for same nozzle size '
+	  planner.flow_percentage[active_extruder+1] = planner.flow_percentage[active_extruder];
+	  #endif
+	  
+	  // Changing tool now
+	  tool_change(active_extruder+1, 0, false); 
+	  
+	  // FwRetract status 
+	  #if ENABLED(FWRETRACT)
+	    fwretract.retracted[active_extruder]=fwretract.retracted[active_extruder-1];
+		fwretract.retracted_swap[active_extruder]=fwretract.retracted_swap[active_extruder-1];	    
+	  #endif
+	  
+	  #if HOTENDS>1
+	    // Heat the new , cold the old   
+	    thermalManager.setTargetHotend(thermalManager.target_temperature[active_extruder-1], active_extruder);
+	    thermalManager.setTargetHotend(0, active_extruder-1);
+	  #endif
+	  
+      resume_print(load_length, ADVANCED_PAUSE_SPOOL_SWAP_EXTRUDE, beep_count);
+	  
+      //Managing if swap next time
+	  if (active_extruder >= (EXTRUDERS-1)) swap_spool_enabled = false ;
+	  if (swap_spool_stop <= active_extruder) swap_spool_enabled = false ;
+      	  
+    }//end next tool
+    else 
+     if (pause_print(retract, z_lift, x_pos, y_pos, unload_length, beep_count, true)) {
+     wait_for_filament_reload(beep_count);
+     resume_print(load_length, ADVANCED_PAUSE_EXTRUDE_LENGTH, beep_count);       
+    }      
```

Comment:
The spool swap secret is here
No comment needed , all is described

Reply:
Please correct indentation.